.github/renovate.json5

```diff
@@ -0,0 +1,91 @@
+{
+  "extends": [
+    "config:recommended",
+    ":approveMajorUpdates",
+  ],
+  // submit on Monday midnight (UTC)
+  "schedule": [
+    "after 12am on Monday"
+  ],
+  "minimumReleaseAge": "14 days",
+  "vulnerabilityAlerts": {
+    "labels": [
+      "type:security"
+    ],
+    "minimumReleaseAge": "1 days"
+  },
+  "labels": ["dependencies"],
+  "dependencyDashboardLabels": ["type: process"],
+  "dockerfile": {
+    "enabled": false
+  },
+  "semanticCommits": true,
+  "pip-compile": {
+    "enabled": true,
+    "fileMatch": ["(^|/)requirements\\.in$"],
+    "lockFileMaintenance": {
+      "enabled": true
+    },
+  },
+  // pip-compile handles requirements.txt updates.
+  "pip_requirements": {
+    "enabled": false
+  },
+  "constraints": {
+    "python": "~=3.11.0"
+  },
+  "ignorePaths": [
+    "demos/**",
+    "security/language-vulns/**"
+  ],
+  "packageRules": [
+    {"matchCategories": ["python"], "addLabels": ["lang: python"]},
+    {"matchCategories": ["java"], "addLabels": ["lang: java"]},
+    {"matchCategories": ["golang"], "addLabels": ["lang: go"]},
+    {"matchCategories": ["js"], "addLabels": ["lang: nodejs"]},
+    {"matchCategories": ["terraform"], "addLabels": ["lang: terraform"]},
+    {
+      "matchUpdateTypes": ["minor", "patch", "digest"],
+      "automerge": true
+    },
+    {
+      "matchDepTypes": ["devDependencies"],
+      "automerge": true
+    },
+    {
+      "matchCategories": ["java"],
+      "matchPackagePatterns": [
+        "io.grpc:(?:protoc-gen-)?grpc-*"
+      ],
+      "groupName": "gRPC packages"
+    },
+    {
+      "matchCategories": ["java"],
+      "matchPackagePatterns": [
+        "^org.apache.beam:"
+      ],
+      "groupName": "Apache Beam packages"
+    },
+    {
+      "matchCategories": ["java"],
+      "matchPackagePatterns": [
+        "^org.eclipse.jetty:"
+      ],
+      "groupName": "Jetty packages"
+    },
+    {
+      "matchCategories": ["java"],
+      "matchPackagePatterns": [
+        "org.springframework.boot"
+      ],
+      "matchCurrentVersion": "!/.*[SNAPSHOT | M]/",
+      "allowedVersions": "<3",
+      "description": "@akitsch: Spring Boot V3 requires Java 17"
```

Comment:
Would it just prevent major upgrades for spring at all including the dashboard? Ideally I do not want them to be there either because we aren't going to drop Java8 support in near future and I want to avoid any confusion.

Reply:
AIUI, it will not make it disappear from the Dependency Dashboard, because that is not something Renovate does. It will appear as "ignored or blocked", just as the current spring 3x update does (https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/issues/234).
The current springboot 3x update is blocked because the PR was closed un-merged. This is the standard way to signal that renovate should ignore an upgrade. No configuration is needed beyond closing the PR without merging.

if it is really important to you to encode this springboot requirement, i'd recommend simplifying this stanza to something more like:
```
"matchCurrentVersion": ">=2.0.0, <3.0.0",
"allowedVersions": "<3.0.0"
```
This is more explicit that the goal is to preserve springboot 2x.